source/code-snippets/usage-examples/changeStream.js

```diff
@@ -21,21 +26,15 @@ async function run() {
       console.log("received a change to the collection: \t", next);
     });
 
-    // use a timeout to ensure the listener is registered before the insertOne
-    // operation is called.
-    await new Promise(resolve => {
-      setTimeout(async () => {
-        await movies.insertOne({
-          test: "sample movie document",
-        });
-        // wait to close `changeStream` after the listener receives the event
-        setTimeout(async () => {
-          resolve(
-            await changeStream.close(() => console.log("closed the change stream"))
-          );
-        }, 1000);
-      }, 1000);
+    await simulateAsyncPause();
+
+    await movies.insertOne({
+      test: "sample movie document"
     });
+
+    await simulateAsyncPause();
+
+    await changeStream.close(() => console.log("closed the change stream"));
```

Comment:
You can either await close or pass a callback not both. We should just await it here, and move the console.log to right after the close call.